pkg/registry/client.go

```diff
@@ -99,6 +119,15 @@ func (c client) GetImageConfig(image string) (*v1.ConfigFile, error) {
 	return cfg, nil
 }
 
+func (c client) HasPushAccess(image string) (bool, error) {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return false, fmt.Errorf("error checking push access: %w", err)
+	}
+	err = remote.CheckPushPermission(ref, c.getAuthHelper(ref), oktetoHttp.DefaultTransport())
```

Comment:
`oktetoHttp.DefaultTransport()` would this work with self-signed certificate instances? I think you need to use `c.getTransport()`